capsules/src/uart.rs

```diff
@@ -55,45 +56,173 @@ pub struct App {
     read_len: usize,
 }
 
-pub static mut WRITE_BUF: [u8; 64] = [0; 64];
-pub static mut READ_BUF: [u8; 64] = [0; 64];
+pub static mut WRITE_BUF0: [u8; 64] = [0; 64];
+pub static mut READ_BUF0: [u8; 64] = [0; 64];
 
-pub struct Console<'a, U: UART> {
-    uart: &'a U,
-    apps: Grant<App>,
+pub static mut WRITE_BUF1: [u8; 64] = [0; 64];
+pub static mut READ_BUF1: [u8; 64] = [0; 64];
+
+pub struct UartDriver<'a, U: 'static + hil::uart::UART> {
+    uarts: &'a mut [&'a mut Uart<'a, U>],
+    apps: [Grant<App>; 2],
+}
+
+impl<'a, U: 'static + hil::uart::UART> UartDriver<'a, U> {
+    pub fn new(
+        uarts: &'a mut [&'static mut Uart<'a, U>],
+        apps: [Grant<App>; 2]
```

Comment:
The canonical way to do this would be to have an array inside each app's Grant, rather than having multiple grant's per app (one per UART, as I believe you're doing here).

None-the-less, even with this method, you could replace the `apps` field with a parameterized type that is a subtype of `AsRef<[Grant<App>]>` which would be instantiated by the board with an array with the same size as the number of UARTs the driver handles.

Reply:
> There are also the static buffers declared in this file, but they could easily be moved to board main and allow the board main to decide which is IMO, how it should be.

This is a separate issue, obviously, but the convention we have is that the board configuration _always_ decides which buffers to use, and capsules provide static ones that are the right size only for convenience (and the board can choose to use them, or not).